source/ui/snapshot-component.cpp

```diff
@@ -49,17 +49,30 @@ QStringList SnapshotComponent::getSnapshotList(const QString& architecture) {
 void SnapshotComponent::addSnapshot(const QString& architectureIdentifier,
                                     const QString& snapshotName,
                                     const std::string& data) {
+  bool snapshotExists =
+      _snapshotMap.contains(architectureIdentifier, snapshotName);
+
   _snapshotDirectory.mkdir(architectureIdentifier);
+
   auto path = snapshotPath(architectureIdentifier, snapshotName);
   Utility::storeToFile(path, data);
-  _snapshotMap.insert(architectureIdentifier, snapshotName);
-  emit snapshotsChanged();
+
+  if (!snapshotExists) {
+    // As the _snapshotMap is a QMultiHash, duplicates of key value pairs can be
+    // inserted, which is stopped by this.
+    _snapshotMap.insert(architectureIdentifier, snapshotName);
+    emit snapshotsChanged();
+  }
 }
 
 void SnapshotComponent::removeSnapshot(const QString& architectureIdentifier,
-                                       const QString& snapshotName) {
-  auto path = snapshotPath(architectureIdentifier, snapshotName);
-  QFile(QString::fromStdString(path)).remove();
+                                       const QString& snapshotName,
+                                       bool removePermanently) {
+  if (removePermanently) {
```

Comment:
So if removePermanently is false, the snapshot is just deleted from the list and not displayed, but loaded again once the simulator is restarted? Is this intended behaviour?